src/serialization/Portable.ts

```diff
@@ -41,127 +48,607 @@ export enum FieldType {
     FLOAT_ARRAY = 17,
     DOUBLE_ARRAY = 18,
     UTF_ARRAY = 19, // Defined for backwards compatibility.
-    STRING_ARRAY = 19
+    STRING_ARRAY = 19,
+    DECIMAL = 20,
+    DECIMAL_ARRAY = 21,
+    TIME = 22,
+    TIME_ARRAY = 23,
+    DATE = 24,
+    DATE_ARRAY = 25,
+    TIMESTAMP = 26,
+    TIMESTAMP_ARRAY = 27,
+    TIMESTAMP_WITH_TIMEZONE,
+    TIMESTAMP_WITH_TIMEZONE_ARRAY = 29
 }
 
 /**
  * Writer helper for {@link Portable} objects.
  */
 export interface PortableWriter {
+
+    /**
+     * Writes a number as 32-bit signed integer.
+     *
+     * @param fieldName name of the field
+     * @param value     int value to be written. The value must be a valid signed 32-bit integer.
+     * Behavior is undefined when value is anything other than a signed 32-bit integer.
```

Comment:
```suggestion
     * Behavior is undefined when value is out of the signed 32-bit integer range.
```
Maybe this one is a bit more clear. WDYT? If you agree with that, we should also update the other ones.

Reply:
Actually, these are directly taken from [Node.js docs.](https://nodejs.org/docs/latest-v14.x/api/buffer.html#buffer_buf_writeint32be_value_offset). I think both versions are fine. The current version has a wider meaning, i.e when value is not a number behaviour is undefined. I think it can stay like this because I have no strong choice.